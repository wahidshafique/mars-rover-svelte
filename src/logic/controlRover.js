export const rotateRover = (command, currentOrientation) => {
  if (command === 'l') {
    if (currentOrientation <= -90) {
      return 180;
    }
    return currentOrientation - 90;
  } else if (command === 'r') {
    if (currentOrientation > 90) {
      return -90;
    }
    return currentOrientation + 90;
  }
};

export const moveRoverOneStep = (
  currentDir,
  gridWidth,
  gridHeight,
  roverX,
  roverY
) => {
  const directionMap = {
    n: { newX: roverX, newY: roverY - 1 },
    e: { newX: roverX + 1, newY: roverY },
    s: { newX: roverX, newY: roverY + 1 },
    w: { newX: roverX - 1, newY: roverY },
  };
  // based on our orientation, move the rover accordingly
  const { newX, newY } = directionMap[currentDir];

  if (newX > gridWidth - 1 || newX < 0 || newY > gridHeight - 1 || newY < 0) {
    console.log('cannot move any more in this direction ', roverX, roverY);
    // set it to what it already is
    return { newX: roverX, newY: roverY };
  }

  return { ...directionMap[currentDir] };
};
