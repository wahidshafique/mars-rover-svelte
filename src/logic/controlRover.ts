import type { RoverDirection } from './types';

export const rotateRover = (command: 'l' | 'r', currentAngle: number) => {
  if (command === 'l') {
    if (currentAngle <= -90) {
      return 180;
    }
    return currentAngle - 90;
  } else if (command === 'r') {
    if (currentAngle > 90) {
      return -90;
    }
    return currentAngle + 90;
  }
};

export const moveRoverOneStep = (
  currentDir: string | RoverDirection,
  gridWidth: number,
  gridHeight: number,
  roverX: number,
  roverY: number
) => {
  const directionMap = {
    n: { newX: roverX, newY: roverY - 1 },
    e: { newX: roverX + 1, newY: roverY },
    s: { newX: roverX, newY: roverY + 1 },
    w: { newX: roverX - 1, newY: roverY },
  };
  // based on our angle, move the rover accordingly
  const { newX, newY }: { newX: number; newY: number } = directionMap[
    currentDir
  ];

  if (newX > gridWidth - 1 || newX < 0 || newY > gridHeight - 1 || newY < 0) {
    console.log('cannot move any more in this direction ', roverX, roverY);
    // set it to what it already is
    return { newX: roverX, newY: roverY };
  }

  return { ...directionMap[currentDir] };
};
