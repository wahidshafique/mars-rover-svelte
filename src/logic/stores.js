import { writable } from 'svelte/store';

export const grid = writable([[], []]);
export const gridDimensions = writable({
  width: 16,
  height: 9,
});
// TODO: remove after
export const rovers = writable([
  {
    name: 'bob',
    x: 3,
    y: 3,
    orientation: 90,
  },
]);
