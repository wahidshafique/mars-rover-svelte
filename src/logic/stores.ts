import { writable } from 'svelte/store';
import type { Rover, GridDims } from './types';

export const gridDimensions = writable<GridDims>({
  width: 16,
  height: 9,
});
// TODO: remove after
export const rovers = writable<Array<Rover>>([]);
