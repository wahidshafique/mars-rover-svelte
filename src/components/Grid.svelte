<script>
  import Rover from './Rover.svelte';
  import { gridDimensions, rovers } from '../logic/stores';

  import genGrid from '../logic/genGrid';

  $: grid = genGrid($gridDimensions.x, $gridDimensions.y, $rovers);
</script>

<div class="p-4 grid-wrapper">
  {#each grid as _, col}
    <div class="clear-both">
      {#each grid[col] as Item, row}
        <div class="float-left whitespace-no-wrap">
          <div class="relative">
            <img src="{Item.url}" alt="{`tile-${row}-${col}`}" />
            {#if Item.rover}
              <Rover title="{Item.rover.name}" />
            {/if}
          </div>
        </div>
      {/each}
    </div>
  {/each}
</div>
