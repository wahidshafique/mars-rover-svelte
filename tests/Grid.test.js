import { render } from '@testing-library/svelte';
import { gridDimensions } from '../src/logic/stores';
import Grid from '../src/components/Grid.svelte';

export const gridGen = (dims) => {
  gridDimensions.update(() => dims);
  const { getByAltText } = render(Grid);
  // verify each grid cell has the correct position ala its alt-text
  for (let y = 0; y < dims.height; y++) {
    for (let x = 0; x < dims.width; x++) {
      expect(getByAltText(`tile-${x}-${y}`));
    }
  }
};

describe('Grid', () => {
  test('must match snapshot', () => {
    const { container } = render(Grid);

    expect(container).toMatchSnapshot();
  });

  test('must render a 2x2 grid', () => {
    const dims = {
      width: 2,
      height: 2,
    };
    gridGen(dims);
  });

  test('must render a 40x27 grid', () => {
    const dims = {
      width: 40,
      height: 27,
    };
    gridGen(dims);
  });

  test('must render a 0x0 grid', () => {
    const dims = {
      width: 0,
      height: 0,
    };
    gridGen(dims);
  });
});
