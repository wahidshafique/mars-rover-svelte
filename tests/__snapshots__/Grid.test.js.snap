// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid must match snapshot 1`] = `
<body>
  <div>
    <div
      class="p-4 grid-wrapper"
    >
      <div
        class="clear-both"
      >
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-0-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-1-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-2-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-3-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-4-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-5-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-6-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-7-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-8-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-9-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-10-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-11-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-12-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-13-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-14-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-15-0"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
         
      </div>
      <div
        class="clear-both"
      >
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-0-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-1-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-2-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-3-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-4-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-5-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-6-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-7-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-8-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-9-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-10-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-11-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-12-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-13-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-14-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-15-1"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
         
      </div>
      <div
        class="clear-both"
      >
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-0-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-1-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-2-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-3-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-4-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-5-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-6-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-7-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-8-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-9-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-10-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-11-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-12-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-13-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-14-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-15-2"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
         
      </div>
      <div
        class="clear-both"
      >
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-0-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-1-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-2-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-3-3"
              src="mapTile_c.png"
            />
             
            <div
              class="absolute top-0"
            >
              <img
                alt="rover"
                class="ml-3 mt-3"
                src="rover.png"
                style="transform: rotate(90deg)"
                title="bob-3-3-w"
              />
            </div>
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-4-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-5-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-6-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-7-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-8-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-9-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-10-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-11-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-12-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-13-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-14-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-15-3"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
         
      </div>
      <div
        class="clear-both"
      >
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-0-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-1-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-2-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-3-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-4-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-5-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-6-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-7-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-8-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-9-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-10-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-11-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-12-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-13-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-14-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-15-4"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
         
      </div>
      <div
        class="clear-both"
      >
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-0-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-1-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-2-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-3-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-4-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-5-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-6-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-7-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-8-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-9-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-10-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-11-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-12-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-13-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-14-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-15-5"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
         
      </div>
      <div
        class="clear-both"
      >
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-0-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-1-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-2-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-3-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-4-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-5-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-6-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-7-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-8-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-9-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-10-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-11-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-12-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-13-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-14-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-15-6"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
         
      </div>
      <div
        class="clear-both"
      >
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-0-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-1-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-2-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-3-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-4-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-5-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-6-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-7-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-8-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-9-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-10-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-11-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-12-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-13-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-14-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-15-7"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
         
      </div>
      <div
        class="clear-both"
      >
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-0-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-1-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-2-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-3-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-4-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-5-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-6-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-7-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-8-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-9-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-10-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-11-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-12-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-13-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-14-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
        <div
          class="float-left whitespace-no-wrap"
        >
          <div
            class="relative"
          >
            <img
              alt="tile-15-8"
              src="mapTile_c.png"
            />
             
          </div>
        </div>
         
      </div>
    </div>
  </div>
</body>
`;
